src/serac/physics/utilities/weak_form/tests/tensor_unit_tests_cuda.cu

```diff
@@ -0,0 +1,129 @@
+
+}
+
+#if 0
+void elasticity_tests() {
+  static constexpr auto abs = [](auto x) { return (x < 0) ? -x : x; };
+
+  static constexpr double lambda = 5.0;
+  static constexpr double mu     = 3.0;
+  static constexpr tensor C      = make_tensor<3, 3, 3, 3>([&](int i, int j, int k, int l) {
+    return lambda * (i == j) * (k == l) + mu * ((i == k) * (j == l) + (i == l) * (j == k));
+  });
+
+  static constexpr auto I     = Identity<3>();
+  constexpr auto        sigma = [=](auto epsilon) { return lambda * tr(epsilon) * I + 2.0 * mu * epsilon; };
+
+  constexpr tensor grad_u = make_tensor<3, 3>([](int i, int j) { return i + 2.0 * j; });
+
+  static_assert(abs(sqnorm(ddot(C, sym(grad_u)) - sigma(sym(grad_u)))) < 1.0e-16);
+
+  constexpr auto epsilon = sym(make_dual(grad_u));
+
+  [[maybe_unused]] static constexpr tensor stress = sigma(epsilon);
+
+  for_constexpr<3, 3>([&](auto i, auto j) { static_assert(abs(sqnorm(C[i][j] - stress[i][j].gradient)) < 1.0e-16); });
+}
+
+void navier_stokes_tests()
+{
+  [[maybe_unused]] static constexpr auto abs = [](auto x) { return (x < 0) ? -x : x; };
+
+  static constexpr auto   I     = Identity<3>();
+  static constexpr double rho   = 3.0;
+  static constexpr double mu    = 2.0;
+  constexpr auto          sigma = [](auto p, auto v, auto L) { return rho * outer(v, v) + 2.0 * mu * sym(L) - p * I; };
+
+  constexpr auto dsigma_dp = [](auto /*p*/, auto /*v*/, auto /*L*/) { return -1.0 * I; };
+
+  constexpr auto dsigma_dv = [&](auto /*p*/, auto v, auto /*L*/) {
+    return make_tensor<3, 3, 3>([&](int i, int j, int k) { return rho * ((i == k) * v[j] + (j == k) * v[i]); });
+  };
+
+  constexpr auto dsigma_dL = [&](auto /*p*/, auto /*v*/, auto /*L*/) {
+    return make_tensor<3, 3, 3, 3>(
+        [&](int i, int j, int k, int l) { return mu * ((i == k) * (j == l) + (i == l) * (j == k)); });
+  };
+
+  constexpr double p = 3.14;
+  constexpr tensor v = {{1.0, 2.0, 3.0}};
+  constexpr tensor L = {{{1.0, 2.0, 3.0}, {4.0, 5.0, 6.0}, {7.0, 8.0, 9.0}}};
+
+  {
+    [[maybe_unused]] static constexpr auto exact = dsigma_dp(p, v, L);
+    [[maybe_unused]] static constexpr auto ad    = sigma(make_dual(p), v, L);
+
+    for_constexpr<3, 3>([&](auto i, auto j) { static_assert(abs(exact[i][j] - ad[i][j].gradient) < 1.0e-16); });
+  }
+
+  {
+    [[maybe_unused]] static constexpr auto exact = dsigma_dv(p, v, L);
+    [[maybe_unused]] static constexpr auto ad    = sigma(p, make_dual(v), L);
+
+    for_constexpr<3, 3>([&](auto i, auto j) { static_assert(abs(sqnorm(exact[i][j] - ad[i][j].gradient)) < 1.0e-16); });
+  }
+
+  {
+    [[maybe_unused]] static constexpr auto exact = dsigma_dL(p, v, L);
+    [[maybe_unused]] static constexpr auto ad    = sigma(p, v, make_dual(L));
+
+    for_constexpr<3, 3>([&](auto i, auto j) { static_assert(abs(sqnorm(exact[i][j] - ad[i][j].gradient)) < 1.0e-16); });
+  }
+}
+#endif
+
+int main()
+{
+  basic_tensor_tests<<<1,1>>>();
+  //elasticity_tests();
```

Comment:
Have these just not been implemented yet?

Reply:
The cuda files are in a weird state, since we agreed to "focus on just getting this PR in" when I was in the middle of it.

Reply:
I'm not sure if it's better to just temporarily remove the cuda stuff for clarity, or leave it in with the understanding that it's still a work-in-progress.

Reply:
I'm okay leaving it in for now as long as you put a comment explaining why.